pkg/tinydtls/patches/0001-build-pkg-tinydtls-make-the-usage-of-extended-master.patch

```diff
@@ -0,0 +1,37 @@
+From da363167afd16496228da439df95686839dcb910 Mon Sep 17 00:00:00 2001
+From: mariemC <charradamyriam@gmail.com>
+Date: Mon, 18 Mar 2024 17:06:40 +0100
+Subject: [PATCH] build/pkg/tinydtls: make the usage of extended master secret
+ and renegotiation info configurable
+
+---
+ dtls.c | 6 ++++--
+ 1 file changed, 4 insertions(+), 2 deletions(-)
+
+diff --git a/dtls.c b/dtls.c
+index 33b352f..6b16ebb 100644
+--- a/dtls.c
++++ b/dtls.c
+@@ -75,6 +75,7 @@ LOG_MODULE_DECLARE(TINYDTLS, CONFIG_TINYDTLS_LOG_LEVEL);
+ 
+ #ifdef RIOT_VERSION
+ # include <memarray.h>
++# include "sock_dtls_types.h"
+ 
+ dtls_context_t dtlscontext_storage_data[DTLS_CONTEXT_MAX];
+ memarray_t dtlscontext_storage;
+@@ -664,8 +665,9 @@ static const dtls_user_parameters_t default_user_parameters = {
+        TLS_NULL_WITH_NULL_NULL
+     },
+ #endif /* DTLS_DEFAULT_CIPHER_SUITES */
+-  .force_extended_master_secret = 1,
+-  .force_renegotiation_info = 1,
++
++  .force_extended_master_secret = SOCK_DTLS_FORCE_EXTENDED_MASTER_SECRET,
++  .force_renegotiation_info = SOCK_DTLS_FORCE_RENEGOTIATION_INFO,
```

Comment:
If these are really optional, the macros should be exposed by the package and injected via CFLAGS (i.e., they shouldn't be RIOT-specific macros, definitely not coming only from `sock_dtls_types.h` which exists only on RIOT).